test/Microsoft.DotNet.Tools.Tests.Utilities/Assertions/FileSystemInfoAssertions.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using FluentAssertions;
+using FluentAssertions.Execution;
+
+namespace Microsoft.DotNet.Tools.Test.Utilities
+{
+    public abstract class FileSystemInfoAssertions<TSubject, TAssertions>
```

Comment:
I struggled for a few minutes to figure out if this could be done by inferring the derived type with `this` rather than using generics like this, but have come to the conclusion that this is a pretty useful pattern 👍